Mathlib/RepresentationTheory/Action/Continuous.lean

```diff
@@ -0,0 +1,114 @@
+of `HasForget₂` instances.
+
+-/
+
+universe u v
+
+open CategoryTheory Limits
+
+namespace Action
+
+variable (V : Type (u + 1)) [LargeCategory V] [ConcreteCategory V] [HasForget₂ V TopCat]
+variable (G : MonCat.{u}) [TopologicalSpace G]
+
+instance : HasForget₂ (Action V G) TopCat :=
+  HasForget₂.trans (Action V G) V TopCat
+
+instance (X : Action V G) : MulAction G ((CategoryTheory.forget₂ _ TopCat).obj X) where
+  smul g x := ((CategoryTheory.forget₂ _ TopCat).map (X.ρ g)) x
+  one_smul x := by
+    show ((CategoryTheory.forget₂ _ TopCat).map (X.ρ 1)) x = x
+    simp
+  mul_smul g h x := by
+    show (CategoryTheory.forget₂ _ TopCat).map (X.ρ (g * h)) x =
+      ((CategoryTheory.forget₂ _ TopCat).map (X.ρ h)
+        ≫ (CategoryTheory.forget₂ _ TopCat).map (X.ρ g)) x
+    rw [← Functor.map_comp, map_mul]
+    rfl
+
+/-- For `HasForget₂ V TopCat` a predicate on an `X : Action V G` saying that the induced action on
+the underlying topological space is continuous. -/
+def IsContinuous (X : Action V G) : Prop :=
+  ContinuousSMul G ((CategoryTheory.forget₂ _ TopCat).obj X)
+
+/-- For `HasForget₂ V TopCat`, this is the full subcategory of `Action V G` where the induced
+action is continuous. -/
+def ContAction : Type _ := FullSubcategory (IsContinuous V G)
+
+instance : Category (ContAction V G) :=
+  FullSubcategory.category (IsContinuous V G)
+
+instance : ConcreteCategory (ContAction V G) :=
+  FullSubcategory.concreteCategory (IsContinuous V G)
+
+instance : HasForget₂ (ContAction V G) (Action V G) :=
+  FullSubcategory.hasForget₂ (IsContinuous V G)
+
+instance : HasForget₂ (ContAction V G) V :=
+  HasForget₂.trans (ContAction V G) (Action V G) V
+
+instance : HasForget₂ (ContAction V G) TopCat :=
+  HasForget₂.trans (ContAction V G) (Action V G) TopCat
+
+namespace ContAction
+
+instance : Coe (ContAction V G) (Action V G) where
+  coe X := X.obj
+
+section Discrete
+
+/-- A predicate on an `X : ContAction V G` saying that the topology on the underlying type of `X`
+is discrete. -/
+def IsDiscrete (X : ContAction V G) : Prop :=
+  DiscreteTopology ((CategoryTheory.forget₂ _ TopCat).obj X)
+
+/-- The subcategory of `ContAction V G` where the topology is discrete. -/
+def DiscreteContAction : Type _ := FullSubcategory (IsDiscrete V G)
+
+namespace DiscreteContAction
+
+instance : Category (DiscreteContAction V G) :=
+  FullSubcategory.category (IsDiscrete V G)
+
+instance : ConcreteCategory (DiscreteContAction V G) :=
+  FullSubcategory.concreteCategory (IsDiscrete V G)
+
+instance : HasForget₂ (DiscreteContAction V G) (ContAction V G) :=
+  FullSubcategory.hasForget₂ (IsDiscrete V G)
+
+instance : HasForget₂ (DiscreteContAction V G) TopCat :=
+  HasForget₂.trans (DiscreteContAction V G) (ContAction V G) TopCat
+
+variable {V G}
+
+instance (X : DiscreteContAction V G) :
+    DiscreteTopology ((CategoryTheory.forget₂ _ TopCat).obj X) :=
+  X.property
+
+end DiscreteContAction
+
+end Discrete
+
+end ContAction
```

Comment:
personally I'd have put `ContAction` and `DiscreteContAction` in the root namespace rather than have them be `Action.ContAction` & `Action.ContAction.DiscreteContAction`, but maybe I'm wrong

Reply:
I agree with this.